src/theory/bv/bitblast/proof_bitblaster.cpp

```diff
@@ -16,19 +16,79 @@
 namespace theory {
 namespace bv {
 
-BBProof::BBProof(TheoryState* state) : d_bb(new BBSimple(state)) {}
+std::unordered_map<Kind, PfRule, kind::KindHashFunction>
+    BBProof::s_kindToPfRule = {
+        {kind::CONST_BITVECTOR, PfRule::BV_BITBLAST_CONST},
+        {kind::EQUAL, PfRule::BV_BITBLAST_EQUAL},
+        {kind::BITVECTOR_ULT, PfRule::BV_BITBLAST_ULT},
+        {kind::BITVECTOR_ULE, PfRule::BV_BITBLAST_ULE},
+        {kind::BITVECTOR_UGT, PfRule::BV_BITBLAST_UGT},
+        {kind::BITVECTOR_UGE, PfRule::BV_BITBLAST_UGE},
+        {kind::BITVECTOR_SLT, PfRule::BV_BITBLAST_SLT},
+        {kind::BITVECTOR_SLE, PfRule::BV_BITBLAST_SLE},
+        {kind::BITVECTOR_SGT, PfRule::BV_BITBLAST_SGT},
+        {kind::BITVECTOR_SGE, PfRule::BV_BITBLAST_SGE},
+        {kind::BITVECTOR_NOT, PfRule::BV_BITBLAST_NOT},
+        {kind::BITVECTOR_CONCAT, PfRule::BV_BITBLAST_CONCAT},
+        {kind::BITVECTOR_AND, PfRule::BV_BITBLAST_AND},
+        {kind::BITVECTOR_OR, PfRule::BV_BITBLAST_OR},
+        {kind::BITVECTOR_XOR, PfRule::BV_BITBLAST_XOR},
+        {kind::BITVECTOR_XNOR, PfRule::BV_BITBLAST_XNOR},
+        {kind::BITVECTOR_NAND, PfRule::BV_BITBLAST_NAND},
+        {kind::BITVECTOR_NOR, PfRule::BV_BITBLAST_NOR},
+        {kind::BITVECTOR_COMP, PfRule::BV_BITBLAST_COMP},
+        {kind::BITVECTOR_MULT, PfRule::BV_BITBLAST_MULT},
+        {kind::BITVECTOR_PLUS, PfRule::BV_BITBLAST_PLUS},
+        {kind::BITVECTOR_SUB, PfRule::BV_BITBLAST_SUB},
+        {kind::BITVECTOR_NEG, PfRule::BV_BITBLAST_NEG},
+        {kind::BITVECTOR_UDIV, PfRule::BV_BITBLAST_UDIV},
+        {kind::BITVECTOR_UREM, PfRule::BV_BITBLAST_UREM},
+        {kind::BITVECTOR_SDIV, PfRule::BV_BITBLAST_SDIV},
+        {kind::BITVECTOR_SREM, PfRule::BV_BITBLAST_SREM},
+        {kind::BITVECTOR_SMOD, PfRule::BV_BITBLAST_SMOD},
+        {kind::BITVECTOR_SHL, PfRule::BV_BITBLAST_SHL},
+        {kind::BITVECTOR_LSHR, PfRule::BV_BITBLAST_LSHR},
+        {kind::BITVECTOR_ASHR, PfRule::BV_BITBLAST_ASHR},
+        {kind::BITVECTOR_ULTBV, PfRule::BV_BITBLAST_ULTBV},
+        {kind::BITVECTOR_SLTBV, PfRule::BV_BITBLAST_SLTBV},
+        {kind::BITVECTOR_ITE, PfRule::BV_BITBLAST_ITE},
+        {kind::BITVECTOR_EXTRACT, PfRule::BV_BITBLAST_EXTRACT},
+        {kind::BITVECTOR_REPEAT, PfRule::BV_BITBLAST_REPEAT},
+        {kind::BITVECTOR_ZERO_EXTEND, PfRule::BV_BITBLAST_ZERO_EXTEND},
+        {kind::BITVECTOR_SIGN_EXTEND, PfRule::BV_BITBLAST_SIGN_EXTEND},
+        {kind::BITVECTOR_ROTATE_RIGHT, PfRule::BV_BITBLAST_ROTATE_RIGHT},
+        {kind::BITVECTOR_ROTATE_LEFT, PfRule::BV_BITBLAST_ROTATE_LEFT},
+};
+
+BBProof::BBProof(TheoryState* state,
+                 ProofNodeManager* pnm,
+                 TConvProofGenerator* tcpg)
+    : d_bb(new BBSimple(state)), d_pnm(pnm), d_tcpg(tcpg)
+{
+}
+
+BBProof::~BBProof() {}
 
 void BBProof::bbAtom(TNode node)
 {
   std::vector<TNode> visit;
   visit.push_back(node);
   std::unordered_set<TNode, TNodeHashFunction> visited;
+
+  bool fine_proofs =
```

Comment:
caml case?

Reply:
We only require camelCase for class/struct members.